### PR TITLE
Session new controller

### PR DESCRIPTION
**Pull request by Steven-Klavins**

### CODE REVIEW

**Comment by FayeCarter** [comment]
it is good. Well done

File: app/controllers/api/v1/registrations_controller.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
+class Api::V1::RegistrationsController < ApplicationController
+   
+    def create
+        user = User.create!(
+          email: params["email"],
+          first_name: params["first_name"],
+          last_name: params["last_name"],
+          password: params["password"]
+        )
```
Comment (tommyrharper) [inline app/controllers/api/v1/registrations_controller.rb:9]: incredible code

File: app/controllers/api/v1/sessions_controller.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+         session[:user_id] = @user.id 
+
+         session[:user_id] = @user.id
+          render json: {
+            status: :created,
+            user: @user,
+            logged_in: true
+          }
+       else
+        render json: {
```
Comment (tommyrharper) [inline app/controllers/api/v1/sessions_controller.rb:14]: nice json

File: Gemfile.lock [hunk 3]
```diff
@@ -117,6 +124,8 @@ GEM
       json
       websocket (~> 1.0)
     rack (2.2.2)
+    rack-cors (1.1.1)
+      rack (>= 2.0.0)
```
Comment (FayeCarter) [inline Gemfile.lock:128]: nice rack

File: spec/controllers/sessions_controller_spec.rb [hunk 4]
```diff
@@ -1,5 +1,48 @@
...
 
-# RSpec.describe SessionsController, type: :controller do
+RSpec.describe Api::V1::SessionsController, type: :controller do
 
-# end
+  before(:each) do
+    @user = User.create(
+    first_name: "Sam",
+    last_name: "Gangi",
+    email: "samgangi@lotr.com",
```
Comment (tommyrharper) [inline spec/controllers/sessions_controller_spec.rb:9]: I'm loving it, especially the LOTR references, superb

File: spec/controllers/sessions_controller_spec.rb [hunk 5]
```diff
@@ -1,5 +1,48 @@
...
+
+    expect(json['status']).to eq "created"
+    expect(json['user']['email']).to eq @user.email
+    expect(json['user']['first_name']).to eq @user.first_name
+    expect(json['user']['last_name']).to eq @user.last_name
+    expect(json['user']['id']).to eq @user.id
+    expect(json['logged_in']).to eq true
+  end     
+
+  it "POST /sessions incorrect email, responds with json status 401 " do
```
Comment (tommyrharper) [inline spec/controllers/sessions_controller_spec.rb:29]: Glad that you bothered to test

File: spec/spec_helper.rb [hunk 6]
```diff
@@ -115,3 +116,19 @@
   Kernel.srand config.seed
 =end
 end
+
+class TestParty
+  include HTTParty
```
Comment (tommyrharper) [inline spec/spec_helper.rb:121]: Great use of HTTP Party

**Review by FayeCarter — APPROVED** [review]

**Review by FayeCarter — APPROVED** [review]

**Pull request merged**